.github/workflows/build-firmware.yaml

```diff
@@ -3,8 +3,54 @@ name: Firmware CI
 on: [push,pull_request]
 
 jobs:
-  build:
+  config:
     runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      # We can tell if this is a push rather than a pull request by testing for the existence of the 'pusher' object
+      # We have to do this on each step in this job rather than the whole job, or everything that needs this job will be skipped.
+      if: ${{ github.event.pusher }}
+      # Because we are making changes, we need to fetch all history
+      with:
+        fetch-depth: 0
+
+    - name: Generate Configs
+      if: ${{ github.event.pusher }}
```

Comment:
PRs still need to generate config, just not commit it

Reply:
Also if we always generate config before building, then we don't care about the execution order.  The generated files can be stale, but that's fine since they'll get generated before building.

Reply:
Guess that's true. Any idea how to do that? I guess the checkout would need to be shared across jobs, but I don't think that works.

Reply:
Just skip the commit/push on PR.  So remove line 19, but leave the commit/push logic as-is.

Reply:
The repo is checked out again in the build jobs, so changes made in this build job won't exist in those jobs.

Reply:
unless you add a generate step as part of the firmware/test builds